src/nn/Execution.js

```diff
@@ -82,7 +82,28 @@ export default class Execution {
    * Schedule evaluation of the execution.
    */
   async startCompute() {
-    await this._preparedModel.execute(this._inputs, this._outputs);
+    let input_shape = this._model._operands[this._inputs.get(0).index].dimensions;
+    if (input_shape.length === 4){
+      let output_shape = this._model._operands[this._outputs.get(0).index].dimensions;
+      let buffer = this._inputs.get(0).buffer;
+      let input_size = input_shape[1] * input_shape[2] * input_shape[3];
+      let output_size = output_shape[1] * output_shape[2] * output_shape[3];
+      let inputIndex = this._inputs.get(0).index;
+      let outputIndex = this._outputs.get(0).index;
+      let outputBuffer = [];
+      let tmp_inputs = new Map();
+      let tmp_outputs = new Map();
+      for(let i=0;i < input_shape[0]; ++i){
+        tmp_inputs.set(0,{index: inputIndex, buffer: buffer.slice(input_size * i,input_size * (i+1))});
+        tmp_outputs.set(0,{index: outputIndex, buffer: buffer.slice(output_size * i,output_size * (i+1))});
+        await this._preparedModel.execute(tmp_inputs, tmp_outputs);
```

Comment:
do we need this code for wasm path? #95 is for WebGL2, did we test wasm?

Reply:
no,wasm support the 4D shapes.
I have optimized the code.